robot/Cumulus/resources/NPSP.py

```diff
@@ -1279,4 +1286,50 @@ def verify_toast_message(self,value):
                 found=True
                 print("Toast message verified")
                 break
-        assert found, "Expected Toast message not found on page"
+        assert found, "Expected Toast message {} not found on page".format(value)
+
+    def edit_record_field_value(self,field,value):
+        """Scrolls just a little below the field
+           Clicks on Edit icon next to field and enters a value into the field"""
+        scroll_loc=npsp_lex_locators["span_button"].format(field)
+        self.selenium.scroll_element_into_view(scroll_loc)
+        self.selenium.execute_javascript("window.scrollBy(0,50)")
+        btn="Edit "+field
+        self.selenium.click_button(btn)
+        footer=npsp_lex_locators["record"]["footer"]
+        self.selenium.wait_until_page_contains_element(footer)
+        self.salesforce.populate_lookup_field(field,value)
+    
+    def edit_record_checkbox(self,field,status):
+        """Scrolls just a little below the field
+           Clicks on Edit icon next to field
+           checks if status is 'checked' 
+           unchecks if status in 'unchecked'"""
+        scroll_loc=npsp_lex_locators["span_button"].format(field)
+        self.selenium.scroll_element_into_view(scroll_loc)
+        self.selenium.execute_javascript("window.scrollBy(0,50)")
+        btn="Edit "+field
+        self.selenium.click_button(btn)
+        footer=npsp_lex_locators["record"]["footer"]
+        self.selenium.wait_until_page_contains_element(footer)
+        self.set_checkbutton_to(field,status)    
+        
+    def save_record(self): 
+        """Saves record by clicking on footer button 'Save'"""
+        footer=npsp_lex_locators["record"]["footer"]
+        self.click_record_button("Save")
+        self.selenium.wait_until_page_does_not_contain_element(footer)
+        self.selenium.execute_javascript("window.scrollTo(0,0)")   
```

Comment:
It might be good to add a comment as to why you're scrolling the window. 

Other than that, this is another good, robust keyword that does some work (clicking a button) and then waiting before returning. 👍

Reply:
added comment